### DISCUSSION TITLE
Jellyfin - Intel HW Acceleration not

### DISCUSSION THREAD

**Discussion by carlito1979** [alexbelgium/hassio-addons#793] [Q&A] [answered]
Hi there. I have installed the jellyfin container, and it works really well, with the exception that I can't get hardware decoding and encoding on my i3-7300U powered system to work.

Going through the documentation from Jellyfin ([hardware-acceleration/intel/](https://jellyfin.org/docs/general/administration/hardware-acceleration/intel/)) I was able to get it working by installing the missing component.
![Screenshot 2023-04-14 070255](https://user-images.githubusercontent.com/13175554/231858944-5801db13-167b-481d-9d95-776f1f19fc1f.png)

So that means HW decoding/encoding now works perfectly, but whenever the addon restarts I lose this component and need to reinstall it.

Is there a way to have this added into the addon by default, or what could I do to ensure this is always installed in the addon?

**Comment by alexbelgium** [comment, marked as answer]
Hi, have you tried setting the "DOCKER_MOD" option to "linuxserver/mods:jellyfin-opencl-intel" ?

Otherwise, you can simply add you code in the file /config/addons_autoscripts/jellyfin.sh and it will execute at boot (the exact path is described in the log).

**Reply by carlito1979** [reply]
Thanks! The docker mod option apparently is only to enable opencl tone mapping and otherwise has nothing to do with hardware acceleration. I did discover the autoscript route and that successfully installs the missing package each time the addon restarts. I appreciate your quick response and awesome work though!

**Reply by alexbelgium** [reply]
Glad it works!

**Reply by guzior-lps** [reply]
> Thanks! The docker mod option apparently is only to enable opencl tone mapping and otherwise has nothing to do with hardware acceleration. I did discover the autoscript route and that successfully installs the missing package each time the addon restarts. I appreciate your quick response and awesome work though!

Can You write full solution here, what have You done to step by step to make HW transcoding working?
